ts/src/pro/binance.ts

```diff
@@ -905,18 +909,12 @@ export default class binance extends binanceRest {
     }
 
     handleOrderBookSubscription (client: Client, message, subscription) {
-        const defaultLimit = this.safeInteger (this.options, 'watchOrderBookLimit', 1000);
-        // const messageHash = this.safeString (subscription, 'messageHash');
         const symbolOfSubscription = this.safeString (subscription, 'symbol'); // watchOrderBook
         const symbols = this.safeValue (subscription, 'symbols', [ symbolOfSubscription ]); // watchOrderBookForSymbols
-        const limit = this.safeInteger (subscription, 'limit', defaultLimit);
         // handle list of symbols
         for (let i = 0; i < symbols.length; i++) {
             const symbol = symbols[i];
-            if (symbol in this.orderbooks) {
-                delete this.orderbooks[symbol];
-            }
-            this.orderbooks[symbol] = this.orderBook ({}, limit);
+            this.tryInitializeOrderBook (symbol, subscription);
```

Comment:
Shouldn't we keep the delete here? What if the subscriptions happens and there is an old cache in this.orderbooks from a previous subscription?